packages/appstore/src/components/modals/wallets-migration-failed/wallets-migration-failed.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import { DesktopWrapper, Text, MobileWrapper, useOnClickOutside, Modal, Button } from '@deriv/components';
+import { localize } from '@deriv/translations';
+import { useStore, observer } from '@deriv/stores';
+import './wallets-migration-failed.scss';
+
+const WalletsMigrationFailed = observer(() => {
+    const { traders_hub } = useStore();
+    const { is_wallet_migration_failed, toggleWalletsMigrationFailedPopup } = traders_hub;
+
+    const wallets_migration_failed_ref = React.useRef<HTMLDivElement>(null);
+
+    const handleClose = () => {
+        toggleWalletsMigrationFailedPopup(false);
+    };
+
+    const handLiveChatButtonClick = () => {
+        window.LC_API.open_chat_window();
+        toggleWalletsMigrationFailedPopup(false);
+    };
+
+    const validateClickOutside = (e: MouseEvent) => {
+        return is_wallet_migration_failed && !wallets_migration_failed_ref?.current?.contains(e.target as Node);
+    };
+
+    useOnClickOutside(wallets_migration_failed_ref, handleClose, validateClickOutside);
```

Comment:
I honestly think that you don't need `useOnClickOutside` for Modals. Try checking out the Modal component or refer to `real-wallets-upgrade`. It will do it automatically.

Reply:
@aizad-deriv `real-wallets-upgrade` uses `useOnClickOutside`, I checked exactly this modal as reference. I noticed Modal component has own `useOnClickOutside` but it doesn't works for me. I've tested it many-many times and this is only one way I found which works for me

Reply:
Can you replace handleClose with this:
const handleClose = () => {
  toggleWalletsMigrationFailedPopup(prevState => !prevState);
};

Reply:
@aizad-deriv in my opinion we have to set `false` always, because it's function to close modal, not for toggle modal. Hope you understand what I mean. We have to set always `false` , because if we use your suggestion and, imagine, somebody by some magic way calls `handleClose` function when modal is closed then the modal will be opened :)

Reply:
@aizad-deriv Or maybe you suggest to rename function from `toggleWalletsMigrationFailedPopup` to, for example, `setWalletsMigrationFailedPopup` ?

Reply:
I think for the renaming to `setWalletsMigrationFailedPopup` is okay too in the `traders-hub-store`. But I suggest you to insert the function in `wallets-migration-failed.tsx`.

Reply:
@aizad-deriv Ok, I'll rename `toggleWalletsMigrationFailedPopup`, but sorry, I'll not change `handleClose` function because we always have to set `false`, not toggling